fog/report/validation/src/lib.rs

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2018-2020 MobileCoin Inc.
+
+#![cfg_attr(not(any(test, feature = "std")), no_std)]
+
+extern crate alloc;
+
+use alloc::string::String;
+use core::fmt::{Debug, Display};
+use mc_account_keys::PublicAddress;
+use mc_crypto_keys::RistrettoPublic;
+
+#[cfg(any(test, feature = "std"))]
+use mockall::*;
+
+pub mod ingest_report;
+
+/// Represents a fog public key validated to use for creating encrypted fog hints.
+/// This object should be constructed only when the IAS report has been validated.
+#[derive(Clone, Debug, Hash, Eq, PartialEq, Ord, PartialOrd)]
+pub struct IasValidatedFogPubkey(pub RistrettoPublic);
+
+impl AsRef<RistrettoPublic> for IasValidatedFogPubkey {
+    fn as_ref(&self) -> &RistrettoPublic {
+        &self.0
+    }
+}
+
+/// Fully resolves a public address to a fully validated fog public key structure,
+/// including all the data from the report server.
+/// This interface may include grpc and so likely cannot be implemented in a way
+/// that is safe for libmobilecoin / asynchronous java requirements.
+#[cfg_attr(any(test, feature = "std"), automock(type Error = String;))]
```

Comment:
I'm not sure if this will work correctly outside this crate, because when other crates are built in `test` they don't get this crate built with `test`. I guess this is why you are adding `std` feature.

If that's all correct, please name the feature `automock` instead to make it more clear what the feature is turning on

Reply:
The feature is needed so that the mock stuff can be enabled by outside crates. I'll rename it to `automock`.